src/wakefs/fuse.cpp

```diff
@@ -148,6 +149,49 @@ static bool collect_result_metadata(const std::string daemon_output, const struc
   return !result_ss.fail();
 }
 
+// Pre-resolve autofs paths through stating the paths on host machine
+static bool resolve_autofs_mounts(const std::vector<mount_op> &mount_ops) {
+  std::vector<std::string> autofs_paths;
+  for (const auto &op : mount_ops) {
+    if (op.type == "autofs-resolve") {
+      autofs_paths.push_back(op.source);
+    }
+  }
+
+  // Sort by path depth as we want parent paths resolved first for nested autofs situations
+  std::sort(autofs_paths.begin(), autofs_paths.end(),
+            [](const std::string &a, const std::string &b) {
+              auto count_directories = [](const std::string &path) {
+                if (path.empty() || path == "/") return 0;
+
+                std::vector<std::string> directories;
+                std::stringstream ss(path);
+                std::string directory;
+
+                while (std::getline(ss, directory, '/')) {
+                  if (!directory.empty()) {  // Skip slashes
+                    directories.push_back(directory);
+                  }
+                }
+
+                return static_cast<int>(directories.size());
+              };
+
+              return count_directories(a) < count_directories(b);
+            });
+
+  for (const auto &path : autofs_paths) {
+    struct stat st;
+    if (stat(path.c_str(), &st) == 0) {
+      std::cerr << "Pre-resolved autofs path: " << path << std::endl;
```

Comment:
Where do debug prints like this land?  If it's runner output then this might be all right, particularly if it's intermixed with similar prints from one or two of the other mount types, but if there's any runners which are tempted to put it in the Job stderr then it would probably be better to remove the temptation by not printing it in the first place.  The failure case is good either way (if more ideal in the runner error).

Also, I recognize that this could easily be run by the `wakebox` tooling or something else external to Wake; we can't just send it to fd4 and call it good.